libtenzir/src/tql2/parser.cpp

```diff
@@ -358,6 +363,9 @@ class parser {
             located{*bin_op, location},
             std::move(right),
           };
+          if (negate) {
+            expr = unary_expr{{unary_op::not_, *negate}, std::move(expr)};
+          }
```

Comment:
I think there is a subtle bug here if `min_prec < 4`, which causes `negate` to not be used.

Reply:
Oh, and you also have to reset it after it has been used.

Reply:
Could you elaborate a little bit on why it won't work for `> 4`?

The negate object is inside the loop so it should be automatically cleaned up.

Reply:
Let `min_prec = 5` and have `not in x` remaining. You set `negate` and `in x` remains. Since `new_prec == 4`, the inner conditional is skipped and you `break` and then return, losing the negation.

Reply:
Added the same check.